src/Declaration.js

```diff
@@ -48,8 +54,10 @@ export default class Declaration {
 	run ( strongDependencies ) {
 		if ( this.tested ) return this.hasSideEffects;
 
-
-		if ( !this.functionNode ) {
+		if ( this.isClassDeclaration && !this.hasConstructor ) {
+			// class declaration without a constructor has no side-effects
+			this.hasSideEffects = false;
```

Comment:
It's there already something else that will look for reassignment of exported bindings, or should we do that here?